cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/LineageHTTPHandler.java

```diff
@@ -115,6 +118,30 @@ public void datasetLineage(HttpRequest request, HttpResponder responder,
                          lineage, getCollapseTypes(collapse)), LineageRecord.class));
   }
 
+  /**
+   * Get the summary for all endpoints in a program run.
+   *
+   * @param namespaceId the namespace in which the pipeline was created
+   * @param appName the appID
+   * @param programType the name of the workflow
+   * @param programName the name of the workflow
+   * @param runId the program run ID
+   */
+  @Beta
+  @GET
+  @Path("/namespaces/{namespace-id}/apps/{app-name}/{program-type}/{program-name}/runs/{run-id}/endpoints")
+  public void getEndpointSummary(HttpRequest request, HttpResponder responder,
+                           @PathParam("namespace-id") String namespaceId,
+                           @PathParam("app-name") String appName,
+                           @PathParam("program-type") String programType,
+                           @PathParam("program-name") String programName,
+                           @PathParam("run-id") String runId) throws Exception {
+    ProgramRunId programRunId = new ProgramRunId(namespaceId, appName, ProgramType.valueOfCategoryName(programType),
```

Comment:
Due to the ongoing lifecycle management changes, we don't want to use ProgramRunId (or any other *Id class) unless it is actually referring to a specific version of the application. 

So in this case, it is better to create a ProgramReference and a RunId, and pass them into getEndpoints() instead of just the single ProgramRunId.